pkg/storegateway/series_refs.go

```diff
@@ -621,7 +711,9 @@ func (s *loadingSeriesChunkRefsSetIterator) Next() bool {
 	loadStats := &queryStats{}
 	defer s.stats.merge(loadStats)
 
-	nextSet := newSeriesChunkRefsSet(len(nextPostings))
+	// This can be released by the caller because loadingSeriesChunkRefsSetIterator doesn't retain it
+	// after Next() will be called again.
+	nextSet := newSeriesChunkRefsSet(len(nextPostings), true)
```

Comment:
Should we also release this when it's empty?

https://github.com/grafana/mimir/blob/4f48f849a5fbb485aa59f7849af9a59ccd539b65/pkg/storegateway/series_refs.go#L738-L740

Reply:
Yes, you're right.